lodash.js

```diff
@@ -3699,8 +3699,13 @@
       return function(number, precision) {
         precision = precision ? toInteger(precision) : 0;
         if (precision) {
-          precision = pow(10, precision);
-          return func(number * precision) / precision;
+          // Shift decimal point with exponential operations
+          number = (number + 'e').split('e'); // Force a split
+          number = func(number[0] + 'e' + (+number[1] + precision)); // Use ToNumber conversions
```

Comment:
I don't know if the end-of-line comments add value here; they were just to explain my deviations from the MDN algorithm. Other than that, LGTM.